tracing.go

```diff
@@ -252,10 +253,26 @@ func ExtractInboundSpan(ctx context.Context, call *InboundCall, headers map[stri
 
 func setPeerHostPort(span opentracing.Span, hostPort string) {
```

Comment:
@prashantv ideally, we would be getting `(host, port)` as proper types, it must be available somewhere in TChannel already. This method is purely internal, so it's rather wasteful to be parsing and converting host:port from string. It's called from the following places, where we have remote peer object, but it only exposes host:port as a string:

```
./tracing.go:157:	setPeerHostPort(span, c.remotePeerInfo.HostPort)
./tracing.go:209:	setPeerHostPort(span, c.remotePeerInfo.HostPort)
./tracing.go:248:		setPeerHostPort(span, call.RemotePeer().HostPort)
```

Another option is to cache it as `map[string]struct{host, hostTag, port}` in some LRU cache.